### PR TITLE
YAML file for CircleCI

### PR DESCRIPTION
**Pull request by GaboGomezT**
## Ticket

PLAT-11232

## Type of PR

- [x] New feature

## Running the demo

In order to run the included demo app, run `yarn run bootstrap` and then `cd demo` and run both `npx react-native run-android` and `npx react-native start` in parallel.

## Review checklist for scaffold changes

- [ ] I have made no scaffold changes - ignore this section questions in that case
- [x] `yarn run template` produces no changes
- [ ] `yarn run raw && yarn run cookie` produces no changes, except for `yarn.lock`
Apparently `yarn run cookie` has issues running on M1 macs, I need someone to validate this runs correctly.
- [x] `yarn run demo` finishes successfully, `demo` directory gets created locally
- [x] **"Running the demo"** opens the app correctly in the welcome screen

## Changes introduced

Created [config.yml](https://github.com/crowdbotics/modules/blob/PLAT-11232-create-validation/.circleci/config.yml) that builds iOS and Android apps validating for any dependency issues.  

## Test and review

Validate the CircleCI pipeline for this PR runs correctly

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,111 @@
+version: 2.1
+jobs:
+  node:
+    working_directory: ~/build
+    docker:
+      - image: cimg/python:3.8.13-node
```
Comment (qcorporation) [inline .circleci/config.yml:6]: So from this website https://circleci.com/developer/images/image/cimg/python?utm_source=google&utm_medium=sem&utm_campaign=sem-google-dg--uscan-en-dsa-tROAS-auth-nb&utm_term=g_-_c__dsa_&utm_content=&gclid=Cj0KCQjw7aqkBhDPARIsAKGa0oIU-5l_QLZWPNB8ebxzbo_O15ynoo8FemKN7WDctz9Nu-vBFKvFmx0aAuEGEALw_wcB 
3.8.13-node has node 16.14.2 
while: https://github.com/crowdbotics/modules/blob/2a211186280210a9dabb104ce9ed021dacec0da7/scaffold/template/custom/.circleci/generate_mobile_android_config.sh#L18 has node 18.16.0

which version of node should we be using. Do the other versions, 
```
build-essential 12.8ubuntu1.1, curl 7.68.0, docker 20.10.12, docker-compose Docker Compose version v2.2.3 v2.2.3, dockerize v0.6.1, git 2.35.1, jq 1.6, node 16.14.2, pip 22.0.4, poetry 1.1.13, pyenv 2.2.5, python2 2.7.18, python3 3.8.13, ubuntu 20.04.3 LTS, virtualenv 20.13.4, wget 1.20.3, wheel 0.37.1, yarn 1.22.5
```

Also match what we are building on the mobile platform?

Comment (mabdullahabid) [reply .circleci/config.yml:6]: 18.16.0. We upgraded our scaffold to use this for our node jobs.

Comment (GaboGomezT) [reply .circleci/config.yml:6]: That's correct @qcorporation. I had issues installing Python to the `cimg/node:18.16.0 ` image so I found this quick solution. I didn't want to take additional time for this and I chose to use `cimg/python:3.8.13-node`. I also thought of creating a new image but that would also take a bit more time. The least required version for node in the scaffold is `16.0` so I assumed this was good enough for now.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+          working_directory: ios
+          timeout: 1200
+
+      - run:
+          name: Populate EnvFile
+          command: env >> .env
+
+      - run:
+          name: Build iOS App
+          command: bundle exec fastlane build_test_dependancies
```
Comment (danielsousaio) [inline .circleci/config.yml:71]:
```suggestion
          command: bundle exec fastlane build_test_dependencies
```

**Review by danielsousaio — CHANGES_REQUESTED** [review]
@GaboGomezT could you please also bump the root `package.json` version and update the `CHANGELOG.md` to talk about this new CircleCI config?

File: .circleci/config.yml (hunk 1 above)
Comment (danielsousaio) [reply .circleci/config.yml:6]: @GaboGomezT do you want to talk about the Python issues when using this image? Perhaps we can debug it a bit.

Comment (GaboGomezT) [reply .circleci/config.yml:6]: Fair enough, I'll work on updating the node version to 18.6.0

**Comment by mabdullahabid** [comment]
@GaboGomezT you need to update the import command to use branch only and remove versions as versions specify the fastlane version not branch/tag version

File: config.js [hunk 3]
```diff
@@ -87,6 +87,30 @@ export default {
...
+        text: "Upgrade my scaffold (2.2.0 -> 2.3.0)",
+        previousVersion: "2.2.0",
+        previousVersionSHA: "99ae211777fe0ae4ae0bd3a601297ad1beb590cb",
+        nextVersion: "2.3.0",
+        nextVersionSHA: "PLAT-11232-create-validation",
+        ignoreTemplatize: [
+          "ios/fastlane/metadata/review_information/review_demo_password.txt",
+          "ios/fastlane/metadata/review_information/review_demo_user.txt"
+        ],
+        upgradeManifestImport: "./manifest/220-to-230.json"
```
Comment (danielsousaio) [inline config.js:113]: This doesn't look right, you need to rebase your PR. Only one new version should be added.

**Review by danielsousaio — COMMENTED** [review]
Please resolve conflicts and rebase your PR when you have a chance.

**Review by mabdullahabid — DISMISSED** [review]

**Review by danielsousaio — APPROVED** [review]

**Pull request merged**